doc/architecture/layers.sql

```diff
@@ -0,0 +1,112 @@
+-- User table
+CREATE TABLE users (
+    ...
+);
+
+-- Project table
+CREATE TABLE map (
+    ...
+);
+
+-- Custom geometry type
+CREATE DOMAIN geom_type AS GEOMETRY
+CHECK (
+    GeometryType(VALUE) IN ('POINT', 'LINESTRING', 'POLYGON') AND
+    ST_SRID(VALUE) = 4326
+);
+
+/*
+    Still up for discussion whether we want to also use cartesian coordinates
+
+    CREATE DOMAIN geom_cartesian_type AS GEOMETRY
+    CHECK (
+        GeometryType(VALUE) IN ('POINT', 'LINESTRING', 'POLYGON') AND
+        ST_SRID(VALUE) = 3857
+    );
+ */
+
+-- Base Layer table
+CREATE TABLE base_layers (
+    id SERIAL PRIMARY KEY,
+    map_id INTEGER REFERENCES map (id),
+    geometry geom_type,
+    base_type TEXT NOT NULL
```

Comment:
What exactly do you mean with base_type?

Reply:
Apologies for the confusion. The base_type column was meant to store textual information about the type of base layer element, such as "sand", "soil", or "rock". Just threw it in there and will remove it.